tests/AssetManagerTest.php

```diff
@@ -577,6 +618,98 @@ public function testAppendTimestamp(): void
         );
     }
 
+    public function testCssAsArrayWithoutUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('Do not set in array CSS URL.');
+        $manager->register([CssAsArrayWithoutUrlAsset::class]);
+    }
+
+    public function testCssAsArrayWithIntegerUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('CSS file should be string. Got integer.');
+        $manager->register([CssAsArrayWithIntegerUrlAsset::class]);
+    }
+
+    public function testCssAsArrayWithEmptyUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('CSS file should be non empty string.');
+        $manager->register([CssAsArrayWithEmptyUrlAsset::class]);
+    }
+
+    public function testJsAsArrayWithoutUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('Do not set in array JS URL.');
+        $manager->register([JsAsArrayWithoutUrlAsset::class]);
+    }
+
+    public function testJsAsArrayWithIntegerUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('JS file should be string. Got integer.');
+        $manager->register([JsAsArrayWithIntegerUrlAsset::class]);
+    }
+
+    public function testJsAsArrayWithEmptyUrl(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('JS file should be non empty string.');
+        $manager->register([JsAsArrayWithEmptyUrlAsset::class]);
+    }
+
+    public function testJsVarAsArrayWithoutName(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('Do not set JavaScript variable name.');
+        $manager->register([JsVarAsArrayWithoutNameAsset::class]);
+    }
+
+    public function testsVarAsArrayWithIntegerName(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage('JavaScript variable name should be string. Got integer.');
+        $manager->register([JsVarAsArrayWithIntegerNameAsset::class]);
+    }
+
+    public function testJsOptionsWithIntegerKey(): void
+    {
+        $manager = $this->createManager();
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage(
+            'JavaScript or CSS options should be list of key/value pairs with string ketys. Got integer key.'
```

Comment:
'JavaScript or CSS options should be list of key/value pairs with string keys. Got integer key.'

Reply:
what is the question?